File: spec/lua/test_imports_abs_rel.lua
local luaunit = require("luaunit")

require("imports_abs_rel")

TestImportsAbsRel = {}

function TestImportsAbsRel:test_imports_abs_rel()
    local r = ImportsAbsRel:from_file("src/fixed_struct.bin")

    luaunit.assertEquals(r.one, 0x50)
    luaunit.assertEquals(r.two.one, 0x41)
    luaunit.assertEquals(r.two.two.one, 0x43)
end

File: spec/python/test_imports_abs_rel.py
import unittest

from imports_abs_rel import ImportsAbsRel

class TestImportsAbsRel(unittest.TestCase):
    def test_imports_abs_rel(self):
        r = ImportsAbsRel.from_file("src/fixed_struct.bin")

        self.assertEqual(r.one, 0x50)
        self.assertEqual(r.two.one, 0x41)
        self.assertEqual(r.two.two.one, 0x43)
